projects/gateway2/proxy_syncer/proxy_syncer.go

```diff
@@ -409,6 +409,9 @@ func (s *ProxySyncer) Init(ctx context.Context, dbg *krt.DebugHandler) error {
 	s.proxyTrigger = krt.NewRecomputeTrigger(true)
 
 	glooProxies := krt.NewCollection(kubeGateways, func(kctx krt.HandlerContext, gw *gwv1.Gateway) *glooProxy {
+		if gw.Spec.GatewayClassName != wellknown.GatewayClassName {
```

Comment:
do we need to consider ExtraGatewayClasses here?

Reply:
Yes, pushed